### DISCUSSION TITLE
unable to edit custom schema from dashboard or js client

### DISCUSSION THREAD

**Discussion by sooqwaifr** [supabase/supabase#3893] [Questions] [answered]
I created a custom schema using:
```
create schema custom_schema;
```
and added the schema to the api schema settings as shown in this link https://supabase.io/docs/reference/javascript/initializing#with-additional-parameters. However, I am unable to add, edit tables or insert rows in this schema  through the dashboard or through js client.
```
const client = createClient(supabaseUrl, supabaseAnonKey, { schema: "custom_schema" });
```

when I use the client on a table it give the following error
```
{
  hint: null,
  message: 'permission denied for table table_name',
  code: '42501',
  details: null
}
```

**Comment by awalias** [comment, marked as answer]
Hi @sooqwaifr thanks for the question!

which SQL client did you use to run `create schema custom_schema;`? Was it inside the supabase dashboard?

it seems like the PostgREST user does not have permissions to access the table, can you try:

```sql
GRANT SELECT, INSERT, UPDATE, DELETE
ON table_name
TO anon; 
```

you can also explore more grants here: https://tableplus.com/blog/2018/04/postgresql-how-to-grant-access-to-users.html

@steve-chavez can you confirm I got this one right?

**Reply by sooqwaifr** [reply]
I used supabase dashboard. I alson added:
```sql
grant usage on schema custom_schema to anon;
```
and 
```sql
GRANT SELECT, INSERT, UPDATE, DELETE
ON table_name
TO anon; 
```
the tables show under the schema but I can't edit or add rows

**Reply by steve-chavez** [reply]
> the tables show under the schema but I can't edit or add rows

After granting permissions, do you still get an error with the JS client?

**Reply by sooqwaifr** [reply]
after granting permissions it seems to works when using the js client, though it doesn't work in the dashboard for editing or insert new record.

**Reply by steve-chavez** [reply]
> after granting permissions it seems to works when using the js client

ok good.

> though it doesn't work in the dashboard for editing or insert new record.

That seems like a bug on the Table Editor, I'd suggest opening an issue so the frontend team can pick it up.

**Reply by Shawns2759** [reply]
I have a very similar issue. Did you ever submit a bug? @steve-chavez
